test/jdk/java/text/Format/DecimalFormat/ToPatternTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8326908
+ * @summary Verify DecimalFormat::toPattern correctness.
+ * @run junit ToPatternTest
+ */
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.Locale;
+import java.util.stream.Stream;
+
+public class ToPatternTest {
+
+    // Ensure that toPattern() provides the correct amount of minimum
+    // and maximum digits for integer/fraction.
+    @ParameterizedTest
+    @MethodSource("minMaxDigits")
+    public void basicTest(int maxInt, int minInt, int maxFrac, int minFrac) {
+        // Use a US dFmt, which uses '.' as the decimal separator
+        DecimalFormat dFmt = (DecimalFormat) NumberFormat.getInstance(Locale.US);
+
+        dFmt.setMaximumIntegerDigits(maxInt);
+        dFmt.setMinimumIntegerDigits(minInt);
+        dFmt.setMaximumFractionDigits(maxFrac);
+        dFmt.setMinimumFractionDigits(minFrac);
+
+        String[] patterns = dFmt.toPattern().split("\\.");
```

Comment:
Instead of assuming/hardcoding the decimal separator, `DecimalFormatSymbols.getInstance(Locale.US).getDecimalSeparator()` may be better.

Reply:
Actually, since `toPattern()` is non-localized, it will always use "." as specified by DecimalFormat.

I removed the misleading comment `// Use a US dFmt, which uses '.' as the decimal separator` that indicated we needed to use a DecimalFormatSymbols where "." was the decimal separator. Can just simply use a DecimalFormat returned by the default constructor.

Reply:
Oh right. Yes, so we should not use the US locale for this test.